### DISCUSSION TITLE
How to kill a working process on closing a terminal tab?

### DISCUSSION THREAD

**Discussion by maxkoshel** [eclipse-theia/theia#9535] [Q&A]
How can I kill a process that runs in a terminal tab that will be closed?

**Comment by msujew** [comment]
@maxkoshel Usually there should be no need for that, since the terminal server will kill the currently running process once the frontend terminal widget is closed as shown in [here](https://github.com/eclipse-theia/theia/blob/144da47ee447017345f447477efc4baf3ba9e65c/packages/terminal/src/node/base-terminal-server.ts#L110-L116).

Do you have an use case where this isn't working as expected?
